spec/compiler/crystal/tools/doc_spec.cr

```diff
@@ -0,0 +1,43 @@
+require "../../../spec_helper"
+
+GIT_REMOTE_PATTERNS = Crystal::Doc::Generator::GIT_REMOTE_PATTERNS.keys
```

Comment:
Ok, so you project the keys. That is not recomputed, but the assert_matches_pattern use .each.compact_map.
I will use `Crystal::Doc::Generator::GIT_REMOTE_PATTERNS.each_key.compact_map ...`  directly.